frontend/src/components/FilesPage/FilesPage.tsx

```diff
@@ -0,0 +1,126 @@
+import { ChangeEventHandler, useState } from 'react';
+import { Navigate } from 'react-router-dom';
+import { MdFolder, MdOutlineInsertDriveFile, MdSearch } from 'react-icons/md';
+import {
+    Cell,
+    ColumnName,
+    FileRow,
+    FolderRow,
+    HeaderRow,
+    IconWrapper,
+    ListContainer,
+    Name,
+    SortIcon,
+    NameButton,
+    InputWithSpace,
+    BreadcrumbsWithSpace,
+} from './services/styled';
+import {
+    by,
+    formatSize,
+    getBreadcumbs,
+    useCurrentDirectory,
+} from './services/helpers';
+import routes, { generateFolderPath } from '../../routes';
+import iconSizes from '../../styleTokens/iconSizes';
+import PageContainer from '../PageContainer/PageContainer';
+
+const FilesPage = () => {
+    const {
+        isLoading,
+        directory: { data, directoryParts },
+    } = useCurrentDirectory();
+
+    const [inputValue, setInputValue] = useState('');
+
+    const [isDescending, setIsDescending] = useState(false);
+
+    if (isLoading) return null;
+
+    if (!data) return <Navigate replace to={routes.notFound} />;
+
+    // NOTE: Sorting and filtering is an expensive operation on big data sets.
+    // Right now, we don't need a memorization since every component update causes a change in options order or number.
+    // Please verify that every time component functionality changes.
+    const sortedItems = [...data.items]
```

Comment:
I don't want to mutate `data.items` (`.sort()` method mutate array)

Reply:
Previously it was needed because `.sort()` mutates array but since I added `.filter()` later `data.items` remain unchanged unchanged. Thanks, nice catch! 15f87acfe6abe26a56c3264d8ef0fdfdf22c0735